public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -295,3 +296,54 @@ export function getStreamSelectorsFromQuery(query: string): string[] {
 
   return labelMatchers;
 }
+
+const PARTITION_LIMIT = 60;
+export function partitionTimeRange(range: TimeRange, unit: DurationUnit = 'm'): TimeRange[] {
```

Comment:
Also, this currently splits 1h query to 60 x 1 minute queries. Is this what we want?

Reply:
The `1` and the `minute` is for testing, because I don't have enough data to cover days locally.

The 60 limit is an arbitrary limit. It can be 30 (for 30 days, should we use days), or whatever else. The point is, without it, you ask for 1 year and it kills the browser. There is a comment in the code explaining this.